tools/json_tools/convert_to_portions.py

```diff
@@ -9,69 +9,28 @@
 args_dict = vars(args.parse_args())
 
 
-class portion_data:
-    encumbrance = -1
-    max_encumbrance = -1
-    coverage = -1
-    covers = []
-    sided = False
-
-
-def portion_null(dat):
-    ret = dat.encumbrance == -1
-    ret = ret and dat.max_encumbrance == -1
-    ret = ret and dat.coverage == -1
-    ret = ret and len(dat.covers) == 0
-    ret = ret and not dat.sided
-    return ret
-
-
-def gen_entry(dat):
-    added = dict()
-
-    if dat.encumbrance != -1:
-        added["encumbrance"] = dat.encumbrance
-        if dat.max_encumbrance != -1:
-            added["encumbrance"] = [dat.encumbrance, dat.max_encumbrance]
-    if dat.coverage != -1:
-        added["coverage"] = dat.coverage
-    if len(dat.covers) != 0:
-        added["covers"] = dat.covers
-    if dat.sided:
-        added["sided"] = True
-
-    return added
+def portionize(jo):
+    dat = dict()
 
+    for member in ["encumbrance", "max_encumbrance",
+                   "coverage", "covers", "sided"]:
+        if member in jo:
+            dat[member] = jo.pop(member)
 
-def portionize(jo):
-    dat = portion_data()
-    if "encumbrance" in jo:
-        dat.encumbrance = jo["encumbrance"]
-        del jo["encumbrance"]
-    if "max_encumbrance" in jo:
-        dat.max_encumbrance = jo["max_encumbrance"]
-        del jo["max_encumbrance"]
-    if "coverage" in jo:
-        dat.coverage = jo["coverage"]
-        del jo["coverage"]
-    if "covers" in jo:
-        dat.covers = jo["covers"]
-        del jo["covers"]
-    if "sided" in jo:
-        dat.sided = jo["sided"]
-        del jo["sided"]
-
-    if portion_null(dat):
+    if not dat:
         return None
 
+    if "max_encumbrance" in dat:
+        if "encumbrance" not in dat:
+            dat["encumbrance"] = 0
+        dat["encumbrance"] = [dat["encumbrance"], dat.pop("max_encumbrance")]
+
     if "armor_portion_data" not in jo:
-        added = gen_entry(dat)
-        jo["armor_portion_data"] = [added]
+        jo["armor_portion_data"] = [dat]
     else:
-        added = gen_entry(dat)
-        added["//0"] = "Autogenned for item with non-portion and portion data"
-        added["//1"] = "Ensure that this works correctly!"
-        jo["armor_portion_data"].insert(0, added)
+        dat["//0"] = "Autogenned for item with non-portion and portion data"
```

Comment:
```suggestion
        dat["//0"] = "Autogenned (by convert_to_portions.py) for item with non-portion and portion data"
```
For anyone looking at the JSON afterward (assuming removal of the second comment line but not this one).

Reply:
I think that anyone who is removing the second comment should also remove the first, it's not useful to keep around after ensuring that it functions correctly.

Reply:
One can think it's working but not actually be, but OK.